### PR TITLE
BaSP-TG-61 : employees create, update & delete with redux

### PR DESCRIPTION
**Pull request by Maxi-SR**
-  Create employee : 
When clicking on the add button in the employees table page, it should navigate to the employee form page. We should complete the form with data and press the confirm button. It should navigate to the employees table and a new row of employee should appear, with all new data.
-  Delete employee : 
In the employees table, when clicking the delete button of a row, a confirmation modal is open to delete an employee. In case of 'No', the modal should be closed and in case of 'Yes', it should delete the selected admin. 
-  Edit an employee : 
When clicking on the edit button inside employees table row, it should navigate to the employees form page. We should check that the form gets all the employee info and we should update all the changes we made. Once confirmed all these changes, it should navigate back to the employees table and it should reflect all changes made.

### CODE REVIEW

File: src/redux/employees/reducer.js [hunk 1]
```diff
@@ -1,21 +1,37 @@
...
 };
 
 const employeesReducer = (state = INITIAL_STATE, action) => {
   switch (action.type) {
     case GET_EMPLOYEES_PENDING:
       return {
         ...state,
-        isPending: true
+        isPending: true,
+        status: 'pending',
```
Comment (Juanoauil) [inline src/redux/employees/reducer.js:33]: Is it necessary to have a status when there is an isPending?

**Review by Juanoauil — CHANGES_REQUESTED** [review]

File: src/Components/Employees/index.js [hunk 2]
```diff
@@ -21,13 +47,54 @@ const Employees = () => {
...
-        <Table data={list} headers={['name', 'lastName', 'phone', 'email', 'password', 'status']} />
+        <Table
+          data={list}
+          headers={['name', 'lastName', 'phone', 'email', 'password', 'status', 'actions']}
+          handleDelete={openDeleteModal}
+          editItem={editEmployee}
+        />
       )}
     </section>
   );
```
[the remark this answers is not in the document]
Comment (Maxi-SR) [reply src/Components/Employees/index.js:100]: done

File: src/Components/Employees/Form/index.js [hunk 3]
```diff
@@ -25,67 +26,38 @@ const AddEmployee = () => {
...
-      if (!error) {
-        toggleModal('success');
-      } else {
-        toggleModal('error');
-      }
-    } catch (error) {
-      alert('Error');
+  useEffect(() => {
+    if (request === 'POST' || request === 'PUT') {
+      toggleModal(status);
```
[the remark this answers is not in the document]
Comment (Maxi-SR) [reply src/Components/Employees/Form/index.js]: done

File: src/redux/employees/reducer.js [hunk 4]
```diff
@@ -1,21 +1,37 @@
...
+        request: 'GET'
```
[the remark this answers is not in the document]
Comment (Maxi-SR) [reply src/redux/employees/reducer.js]: done

File: src/redux/employees/reducer.js (hunk 1 above)
Comment (Maxi-SR) [reply src/redux/employees/reducer.js]: done

**Review by mara666 — APPROVED** [review]
Good job Maxi!

**Review by Juanoauil — APPROVED** [review]

**Comment by EliasBermudez13** [comment]
Looks good to me, nice job its aprooved.

**Pull request merged**